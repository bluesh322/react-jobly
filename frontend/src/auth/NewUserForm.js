import React, { useState } from "react";
import clsx from "clsx";
import { useHistory } from "react-router-dom";
import { Typography, Container, Box, FormControl, TextField, Button, makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    margin: {
        margin: theme.spacing(1),
    },
    withoutLabel: {
        marginTop: theme.spacing(3),
    },
    textField: {
        width: '25ch'
    },
    btn: {
        width: 100,
        marin: theme.spacing(2),
    },
    text: {
        display: 'flex',
    }
}))

const NewUserForm = ({signup}) => {
    const classes = useStyles();
    const INITIAL_STATE = {
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: ""
    }
    const history = useHistory();
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        e.preventDefault();
        const { name, value } = e.target;
        setFormData((data) => ({
            ...data,
            [name]: value,
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        const {username, password, firstName, lastName, email} = formData;
        signup({username, password, firstName, lastName, email});
        setFormData(INITIAL_STATE);
        history.push("/");
    }

    return (
        <Container justifycontent="center" maxWidth="md" fixed>
            <Box className={classes.root} my={2}>
            <Typography className={classes.text} variant="h3">Welcome to jobly!</Typography>
            <form onSubmit={handleSubmit}>
                <FormControl>
                    <TextField id="username" className={clsx(classes.margin, classes.withoutLabel, classes.textField)} variant="outlined" label="username" name="username" value={formData.username} onChange={handleChange}></TextField>
                    <TextField id="password" className={clsx(classes.margin, classes.withoutLabel, classes.textField)} variant="outlined" label="password" name="password" value={formData.password} onChange={handleChange}></TextField>
                    <TextField id="firstName" className={clsx(classes.margin, classes.withoutLabel, classes.textField)} variant="outlined" label="firstName" name="firstName" value={formData.firstName} onChange={handleChange}></TextField>
                    <TextField id="lastName" className={clsx(classes.margin, classes.withoutLabel, classes.textField)} variant="outlined" label="lastName" name="lastName" value={formData.lastName} onChange={handleChange}></TextField>
                    <TextField id="email" className={clsx(classes.margin, classes.withoutLabel, classes.textField)} variant="outlined" label="email" name="email" value={formData.email} onChange={handleChange}></TextField>
                    <Button className={classes.btn} variant="contained" color="primary" type="submit">Sign Up!</Button>
                </FormControl>
            </form>      
            </Box>
        </Container>
    );
}

export default NewUserForm;