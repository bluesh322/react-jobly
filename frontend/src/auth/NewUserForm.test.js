import React from "react";
import { render } from "@testing-library/react";
import NewUserForm from "./NewUserForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <NewUserForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
