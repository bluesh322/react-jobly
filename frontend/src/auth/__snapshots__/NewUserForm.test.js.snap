// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthMd"
    justifycontent="center"
  >
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-root-1"
    >
      <h3
        class="MuiTypography-root makeStyles-text-6 MuiTypography-h3"
      >
        Welcome to jobly!
      </h3>
      <form>
        <div
          class="MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-withoutLabel-3 makeStyles-textField-4"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              username
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-withoutLabel-3 makeStyles-textField-4"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="password"
                name="password"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-withoutLabel-3 makeStyles-textField-4"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="firstName"
              id="firstName-label"
            >
              firstName
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    firstName
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-withoutLabel-3 makeStyles-textField-4"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="lastName"
              id="lastName-label"
            >
              lastName
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="lastName"
                name="lastName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    lastName
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-withoutLabel-3 makeStyles-textField-4"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-10"
                >
                  <span>
                    email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-5 MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Sign Up!
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
